### PR TITLE
Adding Extortion System

### PR DESCRIPTION
**Pull request by djlad**
Hi, I am a long time player and big fan of the game. I would love to contribute if I can. I have professional programming experience, but I haven't worked on any open source projects.

To get familiar with the code, I added a simple extortion system where you can demand money from disabled enemies. I saw this feature in the development roadmap.

Is this a feature that I can try to implement? I implemented the basic feature in this fork, but I would fine tune it if we want to put it in the game.

Added behavior:
- If player hails a disabled enemy ship a demand bribe option is shown
- If ship not heroic or forebearing, the player is given 1 percent of the disabled ship's value

Potential tweaks:
- if player destroys ships that offer bribes they lose reputation
- if player player has low reputation ships will not offer bribes
- ships might only offer bribes if the player has more firepower than ship's faction.

Resolves https://github.com/endless-sky/endless-sky/issues/3949

### CODE REVIEW

**Comment by tehhowch** [comment]
Sure, by all means feel free to propose an implementation. It might be good to first expound how the feature should work, and also to identify any existing discussions of the concept. One such discussion is here: https://github.com/endless-sky/endless-sky/issues/3949

**Comment by Zitchas** [comment]
Sounds like an interesting idea.

**Comment by djlad** [comment]
Ah thanks, I looked through a bunch of the issues, but I didn't see that one. I will take those requirements into account and propose a more detailed system there.

File: source/HailPanel.cpp [hunk 1]
```diff
@@ -286,19 +295,30 @@ bool HailPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, boo
...
 			if(ship)
 			{
-				ship->GetGovernment()->Bribe();
-				Messages::Add("You bribed a " + ship->GetGovernment()->GetName() + " ship "
-					+ Format::Credits(bribe) + " credits to refrain from attacking you today.");
+				if(bribe < 0){
+					// if bribe < 0 player is demanding a bribe
+					if (!ship->GetPersonality().IsHeroic() && !ship->GetPersonality().IsForbearing() && !ship->wasExtorted()){
+						Messages::Add("The captain of the "+ship->Name()+" bribes you "+Format::Credits(-bribe)+" to go away.");
+						message = "Take the money and leave you heathen!";
```
Comment (Pointedstick) [inline source/HailPanel.cpp:302]:
```suggestion
						message = "Take the money and leave me alone!";
```

File: source/HailPanel.cpp [hunk 2]
```diff
@@ -286,19 +295,30 @@ bool HailPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, boo
...
+					} else {
+						message = "You will get nothing from me.";
+						bribe = 0;
+					}
+				}else{
```
Comment (Pointedstick) [inline source/HailPanel.cpp:307]:
```suggestion
				} else {
```

File: source/HailPanel.cpp [hunk 3]
```diff
@@ -286,19 +295,30 @@ bool HailPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, boo
...
+					ship->GetGovernment()->Bribe();
+					Messages::Add("You bribed a " + ship->GetGovernment()->GetName() + " ship "
```
Comment (Pointedstick) [inline source/HailPanel.cpp:309]:
```suggestion
					Messages::Add("You paid a " + ship->GetGovernment()->GetName() + " ship "
```

File: source/HailPanel.cpp [hunk 4]
```diff
@@ -286,19 +295,30 @@ bool HailPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, boo
 		{
 			if(ship)
 			{
-				ship->GetGovernment()->Bribe();
-				Messages::Add("You bribed a " + ship->GetGovernment()->GetName() + " ship "
-					+ Format::Credits(bribe) + " credits to refrain from attacking you today.");
+				if(bribe < 0){
+					// if bribe < 0 player is demanding a bribe
+					if (!ship->GetPersonality().IsHeroic() && !ship->GetPersonality().IsForbearing() && !ship->wasExtorted()){
+						Messages::Add("The captain of the "+ship->Name()+" bribes you "+Format::Credits(-bribe)+" to go away.");
```
Comment (Pointedstick) [inline source/HailPanel.cpp:301]:
```suggestion
						Messages::Add("The captain of the "+ship->Name()+" pays you "+Format::Credits(-bribe)+" to go away.");
```

File: source/HailPanel.cpp [hunk 5]
```diff
@@ -67,6 +67,9 @@ HailPanel::HailPanel(PlayerInfo &player, const shared_ptr<Ship> &ship)
 			if(bribe)
 				message = "If you want us to leave you alone, it'll cost you "
 					+ Format::Credits(bribe) + " credits.";
+		} else {
+			message = "Please don't destroy our ship!";
```
Comment (Pointedstick) [inline source/HailPanel.cpp:71]:
```suggestion
			message = "Please stop attacking us!";
```

**Comment by tehhowch** [comment]
@Pointedstick please see the reference issue for existing discussions on this topic.

I also think it's fine to have both an explicit "demand things from npcs" mechanic and a "NPCs bribe the player by themselves" mechanic.

**Comment by sithrules** [comment]
I am glad to see that this may be added to the game. I would like to add that extorting from a ship should effect the "extorted" Boolean variable not just for that ship, but for the ships in its fleet as well. (As when the player bribes an NPC, their fleet gains the protection as well). Also, extorting merchant ships should maybe cause an increase in reputation with pirates, and possibly extorting ships involved in war should increase the reputation of opposing governments.

**Comment by Pointedstick** [comment]
@djlad, are you interested in resuming the work on this?

**Comment by Zitchas** [comment]
@djlad Are you interested in continuing to work on this?

If not, is anyone else interested in working on this, or should we just close it? I find the concept interesting, but if no-one's going to work on it, then we might as well not have it cluttering up our PR list.

**Comment by Amazinite** [comment]
Closing all PRs which have been marked "orphaned."

**Pull request closed without merging**